File: src/main/kotlin-graphql/io/github/graphql/model/UpdateProjectCardInputTO.kt
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.StringJoiner

@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-06T19:48:04+0800"
)
open class UpdateProjectCardInputTO(
    val clientMutationId: String?,
    val isArchived: Boolean?,
    val note: String?,
    val projectCardId: String
) {

    companion object {
        fun builder(): Builder = Builder()
    }

    // In the future, it maybe change.
    override fun toString(): String {
        val joiner = StringJoiner(", ", "{ ", " }")
        if (clientMutationId != null) {
            joiner.add("clientMutationId: " + GraphQLRequestSerializer.getEntry(clientMutationId))
        }
        if (isArchived != null) {
            joiner.add("isArchived: " + GraphQLRequestSerializer.getEntry(isArchived))
        }
        if (note != null) {
            joiner.add("note: " + GraphQLRequestSerializer.getEntry(note))
        }
        joiner.add("projectCardId: " + GraphQLRequestSerializer.getEntry(projectCardId))
        return joiner.toString()
    }

    class Builder {

        private var clientMutationId: String? = null
        private var isArchived: Boolean? = null
        private var note: String? = null
        private lateinit var projectCardId: String

        fun setClientMutationId(clientMutationId: String?): Builder {
            this.clientMutationId = clientMutationId
            return this
        }

        fun setIsArchived(isArchived: Boolean?): Builder {
            this.isArchived = isArchived
            return this
        }

        fun setNote(note: String?): Builder {
            this.note = note
            return this
        }

        fun setProjectCardId(projectCardId: String): Builder {
            this.projectCardId = projectCardId
            return this
        }

        fun build(): UpdateProjectCardInputTO {
            return UpdateProjectCardInputTO(clientMutationId, isArchived, note, projectCardId)
        }
    }
}

File: src/main/scala-graphql/io/github/graphql/model/UpdateProjectCardInputTO.scala
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import scala.collection.JavaConverters._

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-06T19:48:01+0800"
)
class UpdateProjectCardInputTO(
    clientMutationId: String,
    isArchived: Option[Boolean],
    note: String,
    @javax.validation.constraints.NotNull
    projectCardId: String
) {

    override def toString(): String = {
        Seq(
            if (clientMutationId != null) "clientMutationId: " + GraphQLRequestSerializer.getEntry(clientMutationId) else "",
            if (isArchived.isDefined) "isArchived: " + GraphQLRequestSerializer.getEntry(isArchived.get) else "",
            if (note != null) "note: " + GraphQLRequestSerializer.getEntry(note) else "",
            if (projectCardId != null) "projectCardId: " + GraphQLRequestSerializer.getEntry(projectCardId) else ""
        ).filter(_ != "").mkString("{", ",", "}")
    }
}

object UpdateProjectCardInputTO {

    def builder(): UpdateProjectCardInputTO.Builder = new Builder()

    class Builder {

        private var clientMutationId: String = _
        private var isArchived: Option[Boolean] = _
        private var note: String = _
        private var projectCardId: String = _

        def setClientMutationId(clientMutationId: String): Builder = {
            this.clientMutationId = clientMutationId
            this
        }

        def setIsArchived(isArchived: Option[Boolean]): Builder = {
            this.isArchived = isArchived
            this
        }

        def setNote(note: String): Builder = {
            this.note = note
            this
        }

        def setProjectCardId(projectCardId: String): Builder = {
            this.projectCardId = projectCardId
            this
        }

        def build(): UpdateProjectCardInputTO = new UpdateProjectCardInputTO(clientMutationId, isArchived, note, projectCardId)

    }
}
